src/_P039_Thermosensors.ino

```diff
@@ -506,21 +535,176 @@ boolean Plugin_039(byte function, struct EventStruct *event, String& string)
       }
       else
       {
-        String log;
-        if((log.reserve(66u))) {
-          log = F("P039 :  ");
-          log += getTaskDeviceName(event->TaskIndex);
-          log += F(" : ");
-          log += F("No Sensor attached !");
-          UserVar[event->BaseVarIndex]     = NAN;
-          UserVar[event->BaseVarIndex + 1] = NAN;
-          addLog(LOG_LEVEL_INFO, log);
+        if (loglevelActiveFor(LOG_LEVEL_INFO)) {
+          String log;
+          if((log.reserve(66u))) {
+            log = F("P039 :  ");
+            log += getTaskDeviceName(event->TaskIndex);
+            log += F(" : ");
+            log += F("No Sensor attached !");
+            UserVar[event->BaseVarIndex]     = NAN;
+            UserVar[event->BaseVarIndex + 1] = NAN;
```

Comment:
This is only clearing the UserVar values if loglevel is active.
Probably not the intended result.

Reply:
Not at all... hawk eye !

Reply:
Solved in next push !